File: 3093-sum-of-values-at-indices-with-k-set-bits/3093-sum-of-values-at-indices-with-k-set-bits.go
func bits(num int) int {
    bit := 0
    for num > 0 {
        num &= (num - 1)
        bit++
    }
    return bit
}

func sumIndicesWithKSetBits(nums []int, k int) int {
    sum := 0
    for i := 0; i < len(nums); i++ {
        if bits(i) == k {
            sum += nums[i]
        }
    }
    return sum
}

File: 3093-sum-of-values-at-indices-with-k-set-bits/3093-sum-of-values-at-indices-with-k-set-bits.c
int sumIndicesWithKSetBits(int* nums, int numsSize, int k) {
    int bits(int num) {
        int b = 0;
        while (num) {
            num &= (num - 1);
            b++;
        }
        return b;
    }

    int sum = 0;
    for (int i = 0; i < numsSize; ++i) {
        if (bits(i) == k)
            sum += nums[i];
    }
    return sum;
}